src/NHibernate.Test/NHSpecificTest/NH3023/DeadlockHelper.cs

```diff
@@ -38,6 +40,7 @@ public void ForceDeadlockOnConnection(SqlConnection connection)
 						catch (Exception ex)
 						{
 							winnerEx = ex;
+							winnerLock.Release();
```

Comment:
In case of early failure of the winner thread, the "loser thread" was stuck for 2 minutes. Better release him.